### PR TITLE
Add a note to FAQ about persisting workspace

### PR DESCRIPTION
**Pull request by keynmol**
Following a chat on Discord.

Happy to take notes on the wording.

### CODE REVIEW

File: modules/docs/mdoc/faq.md [hunk 1]
```diff
@@ -96,3 +96,14 @@ https://keys.openpgp.org/search?q=me@scala-steward.org.
...
 
 If you want to switch from the standard INFO log level you can call scala steward with `-DLOG_LEVEL=DEBUG` for example: `scala-steward -DLOG_LEVEL=DEBUG  --workspace  "$STEWARD_DIR/workspace" ...` 
+
+## Why doesn't self-hosted Scala Steward close obsolete PRs?
+
+When a new version of a dependency is found and the PR for older version still exists, Scala Steward 
+will be able to close the old PR.
+
+If you don't observe such behaviour and new version PRs pile up alongside old ones,
+make sure that the workspace folder is persisted between different runs.
```
Comment (mzuehlke) [inline modules/docs/mdoc/faq.md:106]: You  could link `workspace` to the section in `running.md`
https://github.com/keynmol/scala-steward/blob/mention-workspace/modules/docs/mdoc/running.md#workspace

Comment (keynmol) [reply modules/docs/mdoc/faq.md:106]: ah, good idea!

Comment (keynmol) [reply modules/docs/mdoc/faq.md:106]: One question - I see both modules/docs/mdoc/faq.md and docs/faq.md - which one should I update?

Comment (fthomas) [reply modules/docs/mdoc/faq.md:106]: Update `modules/docs/mdoc/faq.md` and then run `sbt docs/mdoc`. `docs/faq.md` is generated from the former, see https://github.com/scala-steward-org/scala-steward/blob/68b7f999125305cfd0a058f89da35999248f406c/build.sbt#L223-L235

File: modules/docs/mdoc/faq.md [hunk 2]
```diff
@@ -96,3 +96,14 @@ https://keys.openpgp.org/search?q=me@scala-steward.org.
...
+
+## Why doesn't self-hosted Scala Steward close obsolete PRs?
+
+When a new version of a dependency is found and the PR for older version still exists, Scala Steward 
+will be able to close the old PR.
+
+If you don't observe such behaviour and new version PRs pile up alongside old ones,
+make sure that the workspace folder is persisted between different runs. 
+
+That's where Steward holds intermediate state about dependency versions which ultimately
```
Comment (fthomas) [inline modules/docs/mdoc/faq.md:108]:
```suggestion
That's where Steward holds intermediate state about dependency versions and created PRs which ultimately
```

**Review by fthomas — APPROVED** [review]
This LGTM, thank you @keynmol!

I only had one small suggestion and would also second @mzuehlke's suggestion.

File: modules/docs/mdoc/faq.md (hunk 1 above)
Comment (keynmol) [reply modules/docs/mdoc/faq.md]: Let me know if https://github.com/scala-steward-org/scala-steward/pull/2886/commits/0fa21c7f33d2b91132b88e31d568e48932c3feff is good enough, happy to iterate :)

**Comment by fthomas** [comment]
I was just wondering if Scala Steward should check at start-up if the workspace is empty and log a warning like: `The workspace is empty. This is expected if this is your first Scala Steward run. Make sure to preserve the workspace between runs for all features to work as expected.`

**Comment by keynmol** [comment]
> I was just wondering if Scala Steward should check at start-up if the workspace is empty and log a warning like: `The workspace is empty. This is expected if this is your first Scala Steward run. Make sure to preserve the workspace between runs for all features to work as expected.`

IMO that'd be a good change.

Initially I scoured the logs to see if there's any suggestion as to why it's not doing what I expected. 

My second thought was - how expensive would an opt-in mode be that gradually rehydrates the workspace on-demand?
What is the main reason why steward needs the workspace persisted vs. querying everything from scratch _if asked to_?

Specifically for this purpose - not for respecting per-repo PR frequency things

File: modules/docs/mdoc/faq.md [hunk 3]
```diff
@@ -96,3 +96,15 @@ https://keys.openpgp.org/search?q=me@scala-steward.org.
...
 
 If you want to switch from the standard INFO log level you can call scala steward with `-DLOG_LEVEL=DEBUG` for example: `scala-steward -DLOG_LEVEL=DEBUG  --workspace  "$STEWARD_DIR/workspace" ...` 
+
+## Why doesn't self-hosted Scala Steward close obsolete PRs?
+
+When a new version of a dependency is found and the PR for older version still exists, Scala Steward 
+will be able to close the old PR.
+
+If you don't observe such behaviour and new version PRs pile up alongside old ones,
+make sure that the [workspace](https://github.com/scala-steward-org/scala-steward/blob/main/modules/docs/mdoc/running.md#workspace)
```
Comment (mzuehlke) [inline modules/docs/mdoc/faq.md:106]:
```suggestion
make sure that the [workspace](running.md#workspace) 
```

File: docs/faq.md [hunk 4]
```diff
@@ -96,3 +96,15 @@ https://keys.openpgp.org/search?q=me@scala-steward.org.
...
 
 If you want to switch from the standard INFO log level you can call scala steward with `-DLOG_LEVEL=DEBUG` for example: `scala-steward -DLOG_LEVEL=DEBUG  --workspace  "$STEWARD_DIR/workspace" ...` 
+
+## Why doesn't self-hosted Scala Steward close obsolete PRs?
+
+When a new version of a dependency is found and the PR for older version still exists, Scala Steward 
+will be able to close the old PR.
+
+If you don't observe such behaviour and new version PRs pile up alongside old ones,
+make sure that the [workspace](https://github.com/scala-steward-org/scala-steward/blob/main/modules/docs/mdoc/running.md#workspace)
```
Comment (mzuehlke) [inline docs/faq.md:106]:
```suggestion
make sure that the [workspace](running.md#workspace) 
```

**Review by mzuehlke — CHANGES_REQUESTED** [review]
The links should be without the path.
See at the top of https://raw.githubusercontent.com/scala-steward-org/scala-steward/main/docs/running.md
the link to `help.md`

**Comment by mzuehlke** [comment]
> My second thought was - how expensive would an opt-in mode be that gradually rehydrates the workspace on-demand?
> What is the main reason why steward needs the workspace persisted vs. querying everything from scratch _if asked to_?

Just an idea, would it be possible to "embed" the relevant information in the created PRs. So the by parsing the PRs the state could be re-created. But I don't know exactly what is persisted.

**Review by mzuehlke — APPROVED** [review]

**Comment by mzuehlke** [comment]
Thank you @keynmol

**Comment by fthomas** [comment]
> > My second thought was - how expensive would an opt-in mode be that gradually rehydrates the workspace on-demand?
> > What is the main reason why steward needs the workspace persisted vs. querying everything from scratch _if asked to_?
> 
> Just an idea, would it be possible to "embed" the relevant information in the created PRs. So the by parsing the PRs the state could be re-created. But I don't know exactly what is persisted.

<details>
<summary>Here is an example of the data Scala Steward stores about a PR.</summary>
<pre>
  "https://github.com/scala-steward-org/scala-steward/pull/2609" : {
    "baseSha1" : "8445342245d2c0e8069744c6b640191dace515d5",
    "update" : {
      "ForGroupId" : {
        "crossDependencies" : [
          [
            {
              "groupId" : "io.circe",
              "artifactId" : {
                "name" : "circe-generic",
                "maybeCrossName" : "circe-generic_2.13"
              },
              "version" : "0.14.1",
              "sbtVersion" : null,
              "scalaVersion" : null,
              "configurations" : null
            }
          ],
          [
            {
              "groupId" : "io.circe",
              "artifactId" : {
                "name" : "circe-generic-extras",
                "maybeCrossName" : "circe-generic-extras_2.13"
              },
              "version" : "0.14.1",
              "sbtVersion" : null,
              "scalaVersion" : null,
              "configurations" : null
            }
          ],
          [
            {
              "groupId" : "io.circe",
              "artifactId" : {
                "name" : "circe-literal",
                "maybeCrossName" : "circe-literal_2.13"
              },
              "version" : "0.14.1",
              "sbtVersion" : null,
              "scalaVersion" : null,
              "configurations" : "test"
            }
          ],
          [
            {
              "groupId" : "io.circe",
              "artifactId" : {
                "name" : "circe-parser",
                "maybeCrossName" : "circe-parser_2.13"
              },
              "version" : "0.14.1",
              "sbtVersion" : null,
              "scalaVersion" : null,
              "configurations" : null
            }
          ],
          [
            {
              "groupId" : "io.circe",
              "artifactId" : {
                "name" : "circe-refined",
                "maybeCrossName" : "circe-refined_2.13"
              },
              "version" : "0.14.1",
              "sbtVersion" : null,
              "scalaVersion" : null,
              "configurations" : null
            }
          ]
        ],
        "newerVersions" : [
          "0.14.2"
        ]
      }
    }
</pre>
</details>

I don't see how we could retrieve the same information from GitHub, GitLab, Bitbucket, or Azure Repos for old PRs. IMO it is just easier to dump this to disk and use it in subsequent runs.

**Pull request merged**